### DISCUSSION TITLE
Serenity BDD architecture

### DISCUSSION THREAD

**Discussion by balivadas** [serenity-bdd/serenity-core#3451] [General]
Does any have have Serenity BDD architecture diagram? if so please share.

**Comment by globalworming** [comment]
What kind of diagram would you like to have? I can you maybe share a similar example to what you are looking for?

**Reply by balivadas** [reply]
@globalworming  thank you for responding.

![SerenityArchitecture_RoughVersion](https://github.com/serenity-bdd/serenity-core/assets/144462165/f3987ada-ab7a-4a5a-9030-14de25b1726c)

Some thing like this. i started working on it. this is just a rough version. Need to beautify it and enhance it. but wanted to check if anyone already has one.

**Reply by globalworming** [reply]
:thinking: i haven't seen anything like this for serenity

also this looks more like a 'process' but an 'architecture' to me, 'execute', 'generate' etc. but it looks useful as a quick "how serenity works" with mvn/junit/cucumber etc. but the actual components  and use cases are much more numerous :)
